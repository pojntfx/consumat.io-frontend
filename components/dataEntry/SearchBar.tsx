import { SearchIcon } from "@heroicons/react/outline";
import { useRouter } from "next/router";
import React, { useState } from "react";

type SearchBarProps = {
  className?: string;
};

function SearchBar({ className }: SearchBarProps) {
  const router = useRouter();
  const { q } = router.query;
  const [query, setQuery] = useState<string>("");

  return (
    <div className={className}>
      <form
        onSubmit={(event) => {
          event.preventDefault();
          router.push({ pathname: "/search", query: { q: query } }, undefined, {
            shallow: true,
          });
        }}
        autoComplete="off"
      >
        <div className="flex flex-row rounded ring-1 ring-inset ring-gray-500 bg-white dark:bg-gray-700">
          <input
            type="search"
            name="q"
            placeholder="Search"
            required
            onChange={(event) => setQuery(event.target.value)}
            className="rounded bg-white dark:bg-gray-700 w-full py-2 pl-4 m-px"
          />

          <button type="submit" className="rounded p-2">
            <SearchIcon className="h-6 w-6 text-gray-500" />
          </button>
        </div>
      </form>
    </div>
  );
}

export default SearchBar;
