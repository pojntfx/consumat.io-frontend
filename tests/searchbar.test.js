import React from "react";
import renderer from "react-test-renderer";
import SearchBar from "../components/dataEntry/SearchBar";
import testExamples from "./testExamples";

const useRouter = jest.spyOn(require("next/router"), "useRouter");

it("renders correctly", () => {
  useRouter.mockImplementationOnce(() => ({
    query: { product: "coffee" },
  }));
  const tree = renderer.create(<SearchBar />).toJSON();
  expect(tree).toMatchSnapshot();
});
