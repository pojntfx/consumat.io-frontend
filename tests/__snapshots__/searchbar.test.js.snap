// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <form
    autoComplete="off"
    onSubmit={[Function]}
  >
    <div
      className="flex flex-row rounded ring-1 ring-inset ring-gray-500 bg-white dark:bg-gray-700"
    >
      <input
        className="rounded bg-white dark:bg-gray-700 w-full py-2 pl-4 m-px"
        name="q"
        onChange={[Function]}
        placeholder="Search"
        required={true}
        type="search"
      />
      <button
        className="rounded p-2"
        type="submit"
      >
        <svg
          className="h-6 w-6 text-gray-500"
          fill="none"
          stroke="currentColor"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M21 21l-6-6m2-5a7 7 0 11-14 0 7 7 0 0114 0z"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeWidth={2}
          />
        </svg>
      </button>
    </div>
  </form>
</div>
`;
